scrapers_next/sc/people.py

```diff
@@ -0,0 +1,148 @@
+from spatula import URL, CSS, HtmlListPage, HtmlPage, XPath, SelectorError
+from openstates.models import ScrapePerson
+import re
+
+
+class LegDetail(HtmlPage):
+    def process_page(self):
+        p = self.input
+
+        cap_addr_path = (
+            XPath(
+                "//*[@id='contentsection']/div[1]/div[3]/h2[contains(text(), 'Columbia Address')]"
+            )
+            .match_one(self.root)
+            .getnext()
+        )
+        cap_addr = cap_addr_path.text
+        cap_addr += " "
+        line2 = cap_addr_path.getchildren()[0].tail
+        if not re.search(r"SC", line2):
+            zipcode = re.search(r"Columbia,?\s\s?(\d{5})", line2).groups()[0]
+            line2 = "Columbia, SC " + zipcode
+        cap_addr += line2
+        p.capitol_office.address = cap_addr
+
+        try:
+            home_addr_path = (
+                XPath(
+                    "//*[@id='contentsection']/div[1]/div[4]/h2[contains(text(), 'Home Address')]"
+                )
+                .match_one(self.root)
+                .getnext()
+            )
+            home_addr = home_addr_path.text
+            home_addr += " "
+            home_line2 = home_addr_path.getchildren()[0].tail
+            if not re.search(r"SC", home_line2):
+                city, h_zip = re.search(r"(.+),?\s(\d{5})", home_line2).groups()
+                home_line2 = city + ", SC " + h_zip
+            home_addr += home_line2
+            p.district_office.address = home_addr
+        except SelectorError:
+            pass
+
+        cap_phones = XPath(
```

Comment:
phew, I see what you mean about being able to simplify phones a bit

I wonder what you think about this approach (I haven't gone deep into the site/data, so maybe this isn't right)

it seems like there is a way to extract each type of phone # and each label, so could you do something where the code builds a dictionary like

phone_numbers = {
   "Capitol Phone": "123-456-7980",
   "Home Phone": "222-222-2222",
   "Business Phone": "444-444-444",
}

which would help detect/avoid dupes, and then we could just read out of that dictionary at the end when assigning to the various offices?

my thinking is that'd separate the code that pulls the various phones from the code that figures out where they should go, and would also help avoid duplicates and all of the checks within nested if statements

glad to talk more if this doesn't seem like it'll work or could use clarification

Reply:
I see what you're saying. I took a crack at this. Looks a little better than previous code in my opinion.